src/com/xilinx/rapidwright/design/tools/ArrayBuilder.java

```diff
@@ -778,10 +774,12 @@ public static void main(String[] args) {
 
         if (ab.getOutOfContext()) {
             // Automatically find bounding PBlock based on used Slices, DSPs, and BRAMs
-            Set<Site> usedSites = array.getUsedSites().stream().filter(
-                    (Site s) -> Arrays.asList(SiteTypeEnum.SLICEL, SiteTypeEnum.SLICEM, SiteTypeEnum.DSP58_PRIMARY,
-                                    SiteTypeEnum.RAMB36, SiteTypeEnum.RAMB18_L, SiteTypeEnum.RAMB18_U)
-                            .contains(s.getPrimarySiteType().getTypeEnum())).collect(Collectors.toSet());
+            Set<Site> usedSites = new HashSet<>();
+            for (SiteInst siteInst : array.getSiteInsts()) {
+                if (isDSP(siteInst) || isBRAM(siteInst) || isSLICE(siteInst)) {
```

Comment:
```suggestion
                if (isSLICE(siteInst) || isBRAM(siteInst) || isDSP(siteInst)) {
```
Since most sites are likely to be SLICE over DSP or BRAM, we should prefer checking for them first as it should be slightly more efficient.